src/Spice86.Core/Emulator/Devices/Timer/Timer.cs

```diff
@@ -31,13 +30,38 @@ public class Timer : DefaultIOPortHandler, ITimeMultiplier, IDebuggableComponent
     /// <summary>
     /// Initializes a new instance of the <see cref="Timer"/> class.
     /// </summary>
-    public Timer(State state, ILoggerService loggerService, DualPic dualPic, CounterConfigurator counterConfigurator, bool failOnUnhandledPort) : base(state, failOnUnhandledPort, loggerService) {
+    public Timer(Configuration configuration, State state, IOPortDispatcher ioPortDispatcher,
+        ILoggerService loggerService, DualPic dualPic) : base(state, configuration.FailOnUnhandledPort, loggerService) {
         _dualPic = dualPic;
-        for (int i = 0; i < _counters.Length; i++) {
-            _counters[i] = new Counter(state,
-                _loggerService,
-                i, counterConfigurator.InstanciateCounterActivator(state));
+        for(int i = 0; i < _counters.Length; i++) {
+            _counters[i] = new Counter(state, loggerService, CreateCounterActivator(state, loggerService, configuration)) {
+                Index = i
+            };
         }
+        InitPortHandlers(ioPortDispatcher);
+    }
+    
+    /// <summary>
+    /// Returns the appropriate <see cref="CounterActivator"/> based on the configuration.
+    /// </summary>
+    /// <param name="state">The CPU registers and flags.</param>
+    /// <param name="loggerService">The service used for logging.</param>
+    /// <param name="configuration">The emulator's configuration.</param>
+    /// <returns>The appropriate <see cref="CyclesCounterActivator"/> or <see cref="TimeCounterActivator"/></returns>
+    private static CounterActivator CreateCounterActivator(State state, ILoggerService loggerService, Configuration configuration) {
```

Comment:
To me it feels a bit strange for a Timer to act as a CounterActivator factory.

Reply:
Fixed.